## 91% of end-end tests have passed
```
Total executed tests: 11
Failed tests: 1
Expected to fail tests: 0
Passed tests: 10
```
```
IDs of failed tests: 703133 
```
<h3>Failed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=fix/sign-tx-biometrics&PR_ID=22040&APK_NAME=22040.apk&TR_CASE_IDS=703133">Rerun failed tests</a></li><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784710">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div># STEP: Check that removed user is not shown in the list anymore</div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:246: in test_restore_multiaccount_with_waku_backup_remove_profile_switch
    self.errors.verify_no_errors()
base_test_case.py:179: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 1: zQ3...dWXh5 was not restored as a contact from waku backup!
E    Device 1: zQ3...Vacac was not restored as a contact from waku backup!
E    Device 1: admin_open was not restored from waku-backup!!
E    Device 1: member_open was not restored from waku-backup!!
E    Device 1: admin_closed was not restored from waku-backup!!
E    Device 1: member_closed was not restored from waku-backup!!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES72oMPFVb8AxoZO9L">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (10)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784709">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESLbc2cOLjaVFtGLoJ">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESPprhvDbQKHXKUpge">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784713">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784716">test_wallet_swap_flow_mainnet</a>, id: 741555 </b></td></tr><tr><td></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784714">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784717">test_wallet_bridge_flow_mainnet</a>, id: 741612 </b></td></tr><tr><td></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784715">test_wallet_send_flow_mainnet</a>, id: 741554 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784707">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES72oMPFVb8AxoZO9L">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784712">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784711">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784708">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESGzoKe5yKonPUgWkI">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESAIijZKRPTAnwRKPM">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/common/standard_authentication/events.cljs
+++ b/src/status_im/common/standard_authentication/events.cljs
@@ -10,16 +10,35 @@
     [utils.re-frame :as rf]
     [utils.security.core :as security]))
 
+(defn- partially-operable-accounts?
+  [accounts]
+  (->> accounts
+       vals
+       (some #(= :partially (:operable %)))
+       boolean))
+
+(rf/reg-event-fx :standard-auth/migrate-partially-operable-accounts
+ (fn [{:keys [db]} [{:keys [masked-password on-success]}]]
+   (let [on-auth-success-callback         #(on-success masked-password)
+         has-partially-operable-accounts? (-> (get-in db [:wallet :accounts])
+                                              partially-operable-accounts?)]
+     {:fx [(if has-partially-operable-accounts?
+             [:dispatch
+              [:wallet/make-partially-operable-accounts-fully-operable
+               {:password   masked-password
+                :on-success on-auth-success-callback
+                :on-error   on-auth-success-callback}]]
+             [:effects.standard-auth/on-auth-success on-auth-success-callback])]})))
+
 (defn- handle-password-success
-  [has-partially-operable-accounts? on-auth-success masked-password]
+  [{:keys [migrate-partially-operable-accounts? on-auth-success masked-password]}]
   (let [on-auth-success-callback #(on-auth-success masked-password)]
     (rf/dispatch [:standard-auth/set-success true])
     (rf/dispatch [:standard-auth/reset-login-password])
-    (if has-partially-operable-accounts?
-      (rf/dispatch [:wallet/make-partially-operable-accounts-fully-operable
-                    {:password   masked-password
-                     :on-success on-auth-success-callback
-                     :on-error   on-auth-success-callback}])
+    (if migrate-partially-operable-accounts?
+      (rf/dispatch [:standard-auth/migrate-partially-operable-accounts
+                    {:masked-password masked-password
+                     :on-success      on-auth-success}])
       (on-auth-success-callback))))
 
 (defn authorize
@@ -42,9 +61,10 @@
                     {:pin        pin
                      :on-success (fn [{:keys [encryption-public-key]}]
                                    (rf/dispatch [:keycard/disconnect])
-                                   (handle-password-success false
-                                                            on-auth-success
-                                                            (security/mask-data encryption-public-key)))
+                                   (handle-password-success {:migrate-partially-operable-accounts? false
+                                                             :on-auth-success on-auth-success
+                                                             :masked-password (security/mask-data
+                                                                               encryption-public-key)}))
                      :on-failure #(rf/dispatch [:keycard/on-action-with-pin-error
                                                 %])}]))}]))}]]
         [:effects.biometric/check-if-available
@@ -85,7 +105,12 @@
         keycard? (get-in db [:profile/profile :keycard-pairing])]
     {:fx [(if keycard?
             [:keychain/get-keycard-keys [key-uid on-auth-success]]
-            [:keychain/get-user-password [key-uid on-auth-success]])
+            [:keychain/get-user-password
+             [key-uid
+              (fn [masked-password]
+                (rf/dispatch [:standard-auth/migrate-partially-operable-accounts
+                              {:masked-password masked-password
+                               :on-success      on-auth-success}]))]])
           [:dispatch [:standard-auth/set-success true]]
           [:dispatch [:standard-auth/reset-login-password]]]}))
 
@@ -108,15 +133,14 @@
 (defn- bottom-sheet-password-view
   [{:keys [on-press-biometric on-auth-success auth-button-icon-left auth-button-label]}]
   (fn []
-    (let [has-partially-operable-accounts? (rf/sub [:wallet/has-partially-operable-accounts?])]
-      [enter-password/view
-       {:on-enter-password   #(handle-password-success
-                               has-partially-operable-accounts?
-                               on-auth-success
-                               (security/hash-masked-password %))
-        :on-press-biometrics on-press-biometric
-        :button-icon-left    auth-button-icon-left
-        :button-label        auth-button-label}])))
+    [enter-password/view
+     {:on-enter-password   #(handle-password-success {:migrate-partially-operable-accounts? true
+                                                      :on-auth-success on-auth-success
+                                                      :masked-password (security/hash-masked-password
+                                                                        %)})
+      :on-press-biometrics on-press-biometric
+      :button-icon-left    auth-button-icon-left
+      :button-label        auth-button-label}]))
 
 (defn authorize-with-keycard
   [_ [{:keys [on-complete]}]]
@@ -154,6 +178,12 @@
  (fn [{:keys [on-close success?]}]
    (when on-close
      (on-close success?))))
+
+(rf/reg-fx
+ :effects.standard-auth/on-auth-success
+ (fn [on-auth-success]
+   (when on-auth-success
+     (on-auth-success))))
 
 (rf/reg-event-fx
  :standard-auth/close

--- a/src/status_im/subs/wallet/wallet.cljs
+++ b/src/status_im/subs/wallet/wallet.cljs
@@ -862,14 +862,6 @@
        fee-formatted))))
 
 (rf/reg-sub
- :wallet/has-partially-operable-accounts?
- :<- [:wallet/accounts]
- (fn [accounts]
-   (->> accounts
-        (some #(= :partially (:operable %)))
-        boolean)))
-
-(rf/reg-sub
  :wallet/accounts-names
  :<- [:wallet/accounts]
  (fn [accounts]

--- a/src/status_im/subs/wallet/wallet_test.cljs
+++ b/src/status_im/subs/wallet/wallet_test.cljs
@@ -942,20 +942,6 @@
           result                (rf/sub [sub-name token-symbol-for-fees])]
       (is (match? result "$0.20")))))
 
-(h/deftest-sub :wallet/has-partially-operable-accounts?
-  [sub-name]
-  (testing "returns false if there are no partially operable accounts"
-    (swap! rf-db/app-db
-      #(assoc-in % [:wallet :accounts] accounts))
-    (is (false? (rf/sub [sub-name]))))
-
-  (testing "returns true if there are partially operable accounts"
-    (swap! rf-db/app-db
-      #(assoc-in %
-        [:wallet :accounts]
-        (update accounts "0x2" assoc :operable :partially)))
-    (is (true? (rf/sub [sub-name])))))
-
 (h/deftest-sub :wallet/zero-balance-in-all-non-watched-accounts?
   [sub-name]
   (testing "returns true if the balance is zero in all non-watched accounts"

--- a/src/tests/integration_test/standard_auth_test.cljs
+++ b/src/tests/integration_test/standard_auth_test.cljs
@@ -17,6 +17,8 @@
 
 (defn auth-success-fixtures
   []
+  (rf/reg-fx :effects.standard-auth/on-auth-success
+   (fn [on-auth-success] (on-auth-success)))
   (rf/reg-fx :effects.biometric/check-if-available
    (fn [{:keys [on-success]}] (on-success)))
   (rf/reg-event-fx :biometric/authenticate
@@ -32,7 +34,7 @@
      (let [on-success-called? (atom false)
            args               (assoc default-args :on-auth-success #(reset! on-success-called? true))]
        (rf/dispatch [:standard-auth/authorize args])
-       (rf-test/wait-for [:standard-auth/on-biometric-success]
+       (rf-test/wait-for [:standard-auth/migrate-partially-operable-accounts]
          (is @on-success-called?))))))
 
 (defn auth-cancel-fixtures

